File: src/main/java/com/dwe/springboot/tools/DownloadHandlerApplication.java
package com.dwe.springboot.tools;


import com.dwe.springboot.tools.car.MigrationService;
import com.dwe.springboot.tools.download.service.FileStorageService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;

@SpringBootApplication
@ConfigurationPropertiesScan
@EnableMethodSecurity
class DownloadHandlerApplication implements CommandLineRunner {

    @Autowired
    FileStorageService storageService;
    @Autowired
    MigrationService migrationService;

    public static void main(String[] args) {
        SpringApplication.run(DownloadHandlerApplication.class, args);
    }

    @Override
    public void run(String... args) {
        migrationService.init();
        // storageService.init();

    }
}

File: src/main/kotlin/com/afolayanseyi/springboot/tutorial/DownloadHandlerApplication.kt
package com.afolayanseyi.springboot.tutorial

import com.afolayanseyi.springboot.tutorial.configuration.StorageProperties
import com.afolayanseyi.springboot.tutorial.service.FileStorageService
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean


@SpringBootApplication
@EnableConfigurationProperties(StorageProperties::class)
class DownloadHandlerApplication {

    @Bean
    fun init(storageService: FileStorageService): CommandLineRunner {
        return CommandLineRunner { args: Array<String?>? ->
               storageService.init()
        }
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(DownloadHandlerApplication::class.java, *args)
        }
    }
}